### PR TITLE
Full review of repository code by Zakala Oleksandr

### PR DESCRIPTION
**Pull request by Rovikido**

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.png" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
```
Comment (Rovikido) [inline client/public/index.html:7]: is unused

File: client/src/App.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from "react";
+import { Route, Routes } from "react-router-dom";
+import { connectWithSocketIOServer } from "./utils/wss";
+import { NotFound } from "./pages/NotFound";
+import { IntroductionPage } from "./pages/Introduction/Introduction";
```
Comment (Rovikido) [inline client/src/App.js:5]: can be imported laizily for faster load time

File: client/src/App.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import JoinRoomPage from "./pages/JoinRoomPage/JoinRoomPage";
+import RoomPage from "./pages/RoomPage/RoomPage";
+import "./App.css";
```
Comment (Rovikido) [inline client/src/App.js:8]: is unused

File: client/src/components/ConnectingButton/ConnectingButton.module.css [hunk 4]
```diff
@@ -0,0 +1,37 @@
+.joinRoomButton {
```
Comment (Rovikido) [inline client/src/components/ConnectingButton/ConnectingButton.module.css:1]: can be consolidated

File: client/src/components/ConnectingButtons/ConnectingButtons.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import ConnectingButton from '../ConnectingButton/ConnectingButton';
+import styles from './ConnectingButtons.module.css';
+
+export const ConnectingButtons = () => {
+  const navigate = useNavigate();
+
+  const pushToJoinRoomPage = () => {
```
Comment (Rovikido) [inline client/src/components/ConnectingButtons/ConnectingButtons.jsx:9]: can be inline function

File: client/src/components/ConnectingButtons/ConnectingButtons.jsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    navigate('/join');
+  };
+
+  const pushToCreateRoomPage = () => {
```
Comment (Rovikido) [inline client/src/components/ConnectingButtons/ConnectingButtons.jsx:13]: can be inline function

File: client/src/index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
```
Comment (Rovikido) [inline client/src/index.js:2]: can import only render

File: client/src/utils/webRTCHandler.js [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+let localStream;
+
+export const getLocalPreviewAndInitRoomConnection = async (
+  isRoomHost,
+  identity,
+  roomId = null
+) => {
+  navigator.mediaDevices
+    .getUserMedia(defaultConstraints)
+    .then((stream) => {
```
Comment (Rovikido) [inline client/src/utils/webRTCHandler.js:23]: async/await can be used instead

File: client/src/utils/webRTCHandler.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    };
+
+    wss.signalPeerData(signalData);
+  });
+
+  peers[connUserSocketId].on("stream", (stream) => {
+    console.log("new stream came");
+
+    addStream(stream, connUserSocketId);
+    streams = [...streams, stream];
```
Comment (Rovikido) [inline client/src/utils/webRTCHandler.js:78]: Set can be used for storing streams. will yield beter performance

File: client/src/utils/wss.js
Comment (Rovikido) [inline client/src/utils/wss.js:1]: error handling may be added

**Review by Rovikido — COMMENTED** [review]
Good code

**Review by Rovikido — COMMENTED** [review]
code review

File: client/public/index.html (hunk 1 above)
Comment (blystsiv) [reply client/public/index.html:7]: ✅

File: client/src/App.js (hunk 3 above)
Comment (blystsiv) [reply client/src/App.js:8]: ✅

File: client/src/utils/webRTCHandler.js (hunk 8 above)
Comment (blystsiv) [reply client/src/utils/webRTCHandler.js:23]: ✅

File: client/src/index.js (hunk 7 above)
Comment (blystsiv) [reply client/src/index.js:2]: ✅

File: client/src/components/ConnectingButtons/ConnectingButtons.jsx (hunk 6 above)
Comment (blystsiv) [reply client/src/components/ConnectingButtons/ConnectingButtons.jsx:13]: ✅

File: client/src/components/ConnectingButtons/ConnectingButtons.jsx (hunk 5 above)
Comment (blystsiv) [reply client/src/components/ConnectingButtons/ConnectingButtons.jsx:9]: ✅

**Pull request merged**